portal-web/test/functional/com/liferay/portalweb/tests/enduser/staging/pgstaging/CPRolesPGStaging.testcase

```diff
@@ -39,251 +47,281 @@ definition {
+				permissionDefinitionValue = "View Staging",
+				roleTitle = "Regrole Name");
 
-		JSONRole.assignRoleToUser(
-			roleTitle = "Regrole Name",
-			userEmailAddress = "userea@liferay.com");
+			JSONRole.assignRoleToUser(
+				roleTitle = "Regrole Name",
+				userEmailAddress = "userea@liferay.com");
+		}
 
-		JSONGroup.assignUserToGroup(
-			groupName = "Site Name",
-			userEmailAddress = "userea@liferay.com");
+		task ("And: the user is assigned to the Site Name group") {
+			JSONGroup.assignUserToGroup(
+				groupName = "Site Name",
+				userEmailAddress = "userea@liferay.com");
+		}
 
-		User.logoutAndLoginPG(
-			password = "test",
-			userLoginEmailAddress = "userea@liferay.com",
-			userLoginFullName = "userfn userln");
+		task ("Then: the user can access the staged page") {
+			User.logoutAndLoginPG(
+				password = "test",
+				userLoginEmailAddress = "userea@liferay.com",
+				userLoginFullName = "userfn userln");
 
-		Navigator.gotoStagedSitePage(
-			pageName = "Staging Test Page",
-			siteName = "Site Name");
+			Navigator.gotoStagedSitePage(
+				pageName = "Staging Test Page",
+				siteName = "Site Name");
 
-		Staging.viewStagingLocalLivePG();
+			Staging.viewStagingLocalLivePG();
 
-		User.logoutAndLoginPG(
-			password = "test",
-			userLoginEmailAddress = "test@liferay.com",
-			userLoginFullName = "test test");
+			User.logoutAndLoginPG(
+				password = "test",
+				userLoginEmailAddress = "test@liferay.com",
+				userLoginFullName = "test test");
+		}
 	}
 
 	@priority = "4"
 	test StagingBlogsRolesWorkflow {
-		JSONUser.addUser(
-			userEmailAddress = "usereapu@liferay.com",
-			userFirstName = "userfnpu",
-			userLastName = "userlnpu",
-			userScreenName = "usersnpu");
-
-		JSONRole.addSiteRole(
-			roleKey = "Power User Name",
-			roleTitle = "Power User Name");
-
-		Permissions.definePermissionViaJSONAPI(
-			resourceAction = "MANAGE_LAYOUTS",
-			resourceName = "com.liferay.portal.kernel.model.Group",
-			roleTitle = "Power User Name",
-			roleType = "site");
-
-		Permissions.definePermissionViaJSONAPI(
-			resourceAction = "VIEW_SITE_ADMINISTRATION",
-			resourceName = "com.liferay.portal.kernel.model.Group",
-			roleTitle = "Power User Name",
-			roleType = "site");
-
-		Permissions.definePermissionViaJSONAPI(
-			resourceAction = "ACCESS_IN_CONTROL_PANEL",
-			resourceName = "com_liferay_layout_admin_web_portlet_GroupPagesPortlet",
-			roleTitle = "Power User Name",
-			roleType = "site");
+		task ("Given: A user is created") {
+			JSONUser.addUser(
+				userEmailAddress = "usereapu@liferay.com",
+				userFirstName = "userfnpu",
+				userLastName = "userlnpu",
+				userScreenName = "usersnpu");
+		}
 
-		Permissions.definePermissionViaJSONAPI(
-			resourceAction = "ADD_ENTRY",
-			resourceName = "com.liferay.blogs",
-			roleTitle = "Power User Name",
-			roleType = "site");
+		task ("And: A power user role is created") {
```

Comment:
I think we could move this to the previous task and have the 2 user creation together. Here, you can define the Permission defining in the task description.